feat: opengraph with nuxt

preschian opened the pull request.

preschian commented:
@SocketSecurity ignore @nuxt/telemetry@2.5.2

preschian commented:
Hi, @kodadot/internal-dev. Sorry, I already deployed this worker to the Canary environment. While waiting for SSR support on our UI, we can now create an open graph component with nuxt in this worker. Once we support SSR in our UI, we can copy the `OgImage` component to our UI

Currently, this worker has already deployed for `ahk`. If it looks smooth, I will open it to another prefix.

In the next phase, I will try to migrate https://github.com/kodadot/workers/blob/main/prerender-opengraph/wrangler.toml and https://github.com/kodadot/workers/blob/main/prerender/wrangler.toml to this worker.

vikiival commented:
Looks good to me, honestly i would 
Move it to separated repo but

Yolo. 

So approved on my side


May @roiLeo can have some feedback

roiLeo commented on nuxt-opengraph/components/OgImage/collection.vue:
```diff
@@ -0,0 +1,62 @@
+  <img :src="image" :alt="title" :style="cover" class="h-full w-full" />
+
+  <div
+    class="flex flex-col justify-end h-full w-full bg-slate-900/85 text-white p-10 text-2xl font-bold absolute inset-0"
```
**question**: how is tailwind imported here?

preschian replied:
already builtin from satori itself https://nuxtseo.com/og-image/guides/using-satori

roiLeo commented on nuxt-opengraph/components/OgImage/collection.vue:
```diff
@@ -0,0 +1,62 @@
+    <h1 class="mb-6 font-bold">{{ title }}</h1>
+    <div class="flex flex-row">
+      <div>
+        <h2 class="font-bold m-0">{{ items }}</h2>
```
**suggestion**: avoid too many h2

roiLeo reviewed (approved):
> Once we support SSR in our UI, we can copy the `OgImage` component to our UI

👍 

> Currently, this worker has already deployed for `ahk`. If it looks smooth, I will open it to another prefix.

open to all prefix and all pages? can we deploy a version only for beta/canary like we do with the other worker? no need for og-image vercel anymore?

> In the next phase, I will try to migrate https://github.com/kodadot/workers/blob/main/prerender-opengraph/wrangler.toml and https://github.com/kodadot/workers/blob/main/prerender/wrangler.toml to this worker.

I'll look into this feel free to tag me for review

vikiival commented on nuxt-opengraph/wrangler.toml:
```diff
@@ -0,0 +1,8 @@
+name = "nuxt-opengraph"
```
```suggestion
name = "ogi"
```

vikiival replied:
Like open graph image :)

roiLeo commented:
wow the preview is amazing!! let's merge it and make it available to other chains

<img width="509" alt="Capture d’écran 2023-11-13 à 9 10 19 AM" src="https://github.com/kodadot/workers/assets/9987732/a966e8eb-bdbc-4d6e-8316-9c351359b91f">

roiLeo commented on nuxt-opengraph/components/OgImage/collection.vue:
```diff
@@ -0,0 +1,62 @@
+  <div
+    class="flex flex-col justify-end h-full w-full bg-slate-900/85 text-white p-10 text-2xl font-bold absolute inset-0"
+  >
+    <img :src="image" :alt="title" class="w-30" />
```
```suggestion
    <img :src="image" :alt="title" class="w-30 rounded-md border border-white" />
```

roiLeo commented on nuxt-opengraph/components/OgImage/collection.vue:
```diff
@@ -0,0 +1,62 @@
+    <img :src="image" :alt="title" class="w-30" />
+    <h1 class="mb-6 font-bold">{{ title }}</h1>
+    <div class="flex flex-row">
+      <div>
+        <h2 class="font-bold m-0">{{ items }}</h2>
+        <h2 class="text-gray-400 m-0">items</h2>
+      </div>
+
+      <!-- <div class="ml-16">
+        <h2 class="font-bold m-0">{{owners}}</h2>
+        <h2 class="text-gray-400 m-0">owners</h2>
+      </div> -->
+
+      <div class="ml-16">
+        <h2 class="font-bold m-0">{{ network }}</h2>
+        <h2 class="text-gray-400 m-0">network</h2>
+      </div>
```
```suggestion
      <div>
        <div class="text-xl font-bold m-0">{{ items }}</div>
        <div class="text-gray-400 m-0">items</div>
      </div>

      <!-- <div class="ml-16">
        <h2 class="font-bold m-0">{{owners}}</h2>
        <h2 class="text-gray-400 m-0">owners</h2>
      </div> -->

      <div class="ml-16">
        <div class="text-xl font-bold m-0">{{ network }}</div>
        <div class="text-gray-400 m-0">network</div>
      </div>
```

roiLeo commented:
Let's go 🚀 !!! Would love to work on other pages!

preschian commented:
> open to all prefix and all pages?

updated, I open it to all prefixes `/collection`

> can we deploy a version only for beta/canary like we do with the other worker?

sure, I will merge it and then deploy


> no need for og-image vercel anymore?

yes, just added a base og-image to the basic layout. endpoint `http://localhost:3000/__og_image__/og.png?image=image_url&text=some_text`

preschian commented:
deployed new ogi version: https://github.com/kodadot/workers/actions/runs/6848197632/job/18617900805
deployed our proxy to canary environment. collection page on canary should be affected: https://github.com/kodadot/workers/actions/runs/6848240845/job/18618030272

https://www.opengraph.xyz/url/https%3A%2F%2Fcanary.kodadot.xyz%2Fbsx%2Fcollection%2F1865909717

og-image vercel replacement endpoint: `https://ogi.kodadot.workers.dev/__og_image__/og.png?image=https://imagedelivery.net/jk5b6spi_m_-9qC4VTnjpg/bafkreiazqh2ba6vrqrtclb5rp4achww3xlo4whdpdu5ce43cdfrvzvokkq/public&text=123%20KSM`
